atmosphere.py

```diff
@@ -0,0 +1,144 @@
+from gpkit import Variable, Model, units, SignomialsEnabled
+import gpkit.tools as gpt
+import numpy as np
+import numpy.testing as npt
+
+class Troposphere(Model):
+    """
+    Density and dynamic viscosity as a function of altitude based on
+    standard atmosphere model for the Troposphere
+
+    Required pressures: downward pressure on rho
+    Assumptions: only valid to the top of the Troposphere (~11km)
+
+    Sources:
+    https://en.wikipedia.org/wiki/Density_of_air#Altitude
+    http://www.digitaldutch.com/atmoscalc/index.htm
+    http://www.engineeringtoolbox.com/standard-atmosphere-d_604.html
+    http://www-mdp.eng.cam.ac.uk/web/library/enginfo/aerothermal_dvd_only/aero/
+           atmos/atmos.html
+    http://www.cfd-online.com/Wiki/Sutherland's_law
+
+    Arguments
+    ---------
+    g : float [m/s^2]
+        Acceleration due to gravity
+    L : float  [K/m]
+        Temperature lapse rate
+    R : float  [J/(kg*K)]
+        Specific gas constant (air)
+    """
+    def __init__(self, g=9.81, L=0.0065, R=287, **kwargs):
+        th = g/(R*L)  # dimensionless
+        self.g, self.L, self.R, self.th = g, L, R, th
+
+        # Free variables
+        h   = Variable('h', 'm', 'Altitude')
+        mu  = Variable("mu", "kg/(m*s)", "Dynamic viscosity")
+        p   = Variable('p', 'Pa', 'Pressure')
+        rho = Variable('\\rho', 'kg/m^3', 'Density')
+        T   = Variable('T', 'K', 'Temperature')
+
+        # Constants
+        C_1  = Variable("C_1", 1.458E-6, "kg/(m*s*K^0.5)",
+                        "Sutherland coefficient")
+        L    = Variable('L', L, 'K/m', 'Temperature lapse rate')
+        p_0  = Variable('p_0', 101325, 'Pa', 'Pressure at sea level')
+        R    = Variable('R', R, 'J/(kg*K)', 'Specific gas constant (air)')
+        T_0  = Variable('T_0', 288.15, 'K', 'Temperature at sea level')
+        T_S  = Variable("T_S", 110.4, "K", "Sutherland Temperature")
+
+        with SignomialsEnabled():
+            objective = rho  # minimize density
+            constraints = [  # Model only valid up to top of the troposphere
+                             h <= 11000*units.m,
+
+                             # Temperature decreases with height at a rate of L
+                             T_0 <= T + L*h,
+
+                             # Pressure-altitude relation
+                             (p/p_0)**(1/th) == T/T_0,
+ 
+                             # Ideal gas law
+                             rho == p/(R*T),
+
+                             # Sutherland viscosity model
+                             ((T+T_S).mono_approximation({T:288.15,
+                                                          T_S:T_S.value.magnitude}))
+                             * mu == C_1 * T**1.5
+                          ]
+        Model.__init__(self, objective, constraints, **kwargs)
```

Comment:
one idea is to leave the objective as `None` here, and only add an objective for testing. It seems to me that atmosphere is a good example of a model that's really a constraint set; the objective here is only being used for testing purposes to apply the required pressure.

@bqpd, `Model.__init__` requires a cost; is the best practice to pass None, or should this `Troposphere` model inherit from `ConstraintSet` instead of Model?